plugins/core/span_tracing.go

```diff
@@ -28,6 +28,11 @@ import (
 )
 
 func NewSegmentSpan(defaultSpan *DefaultSpan, parentSpan SegmentSpan) (s SegmentSpan, err error) {
+	// if current and parent span are both entry span, then use parent span as segment span
+	if defaultSpan.SpanType == SpanTypeEntry && parentSpan != nil && parentSpan.IsEntry() {
```

Comment:
Should we check parent span first before the span creation? It seems to cause a little extra payload?

Reply:
I have advanced the parent check, please re-check